### PR TITLE
Fix loading states when report errors

### PR DESCRIPTION
**Pull request by aaemnnosttv**
## Summary

Addresses issue:

- #4542

## PR Author Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [ ] Run the code.
- [ ] Ensure the acceptance criteria are satisfied.
- [ ] Reassess the implementation with the IB.
- [ ] Ensure no unrelated changes are included.
- [ ] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [ ] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [ ] Ensure the PR has the correct target branch.
- [ ] Double-check that the PR is okay to be merged.
- [ ] Ensure the corresponding issue has a ZenHub release assigned.
- [ ] Add a changelog message to the issue.

### CODE REVIEW

**Comment by aaemnnosttv** [comment]
@kuasha420 looks like Storybook is still broken here according to the check

File: assets/js/modules/analytics/datastore/report.js
Comment (aaemnnosttv) [inline assets/js/modules/analytics/datastore/report.js:64]: Thinking about this again, I think we should address this in our tests/infra where it's failing instead.

For example, in Storybook we should update `fetchMockCatchAll` to return an empty array for this endpoint instead.

I think you said this was raising failures in E2E as well, in which case it should be a matter of updating the relevant intercepted requests to return an array instead.

**Review by aaemnnosttv — COMMENTED** [review]
Thanks @kuasha420 – this is looking good, just a few comments left to resolve; almost there I think 👍 Let me know if you have any other questions.

**Review by aaemnnosttv — COMMENTED** [review]
Thanks @kuasha420 – this looks good, just a few final tweaks to get this over the line such as applying the changes to the tests from the last review consistently and one comment about an intercepted request that I think we want to keep.

I just pushed a few changes that should address these last few things.

**Comment by aaemnnosttv** [comment]
Confirmed that these changes fix the problem testing with permissions errors with Analytics and SC.

**Comment by aaemnnosttv** [comment]
@kuasha420 I can't approve this PR because it's my own 😄 

Can you take a quick look over my changes and approve if they look good to you?

**Review by kuasha420 — APPROVED** [review]
Thank you @aaemnnosttv for all the help with this one! Looks much cleaner now. Nice. :+1:

**Review by tofumatt — COMMENTED** [review]
Looks good to me as well; moving to QA 👍🏻

**Pull request merged**
